libraries/AP_HAL_SITL/sitl_barometer.cpp

```diff
@@ -76,7 +76,9 @@ void SITL_State::_update_barometer(float altitude)
 
     // find data corresponding to delayed time in buffer
     for (uint8_t i=0; i<=baro_buffer_length-1; i++) {
-        time_delta_baro = abs(delayed_time_baro - buffer_baro[i].time); // find difference between delayed time and time stamp in buffer
+        // find difference between delayed time and time stamp in buffer
+        time_delta_baro = abs(
+                (int32_t)(delayed_time_baro - buffer_baro[i].time));
```

Comment:
Wouldn't you need to cast the vars before the subtraction? This is casting after the subtraction right?